app/js/sumo_db.js

```diff
@@ -154,6 +153,21 @@
       endpoint += '?format=json'; // TODO bug 1088014
 
       return request(endpoint, 'GET').then(JSON.parse);
+    },
+
+    update_preference: function(user, setting) {
+      var endpoint = API_V2_BASE + 'user/';
+      endpoint += user.username + '/';
+
+      var delete_setting = endpoint + 'delete_setting/';
+      delete_setting += '?format=json'; // TODO bug 1088014
+      return request_with_auth(delete_setting, 'POST', setting)
+        .then(function(response) {
```

Comment:
We need to be able to set this in one go. Please work with Mike in [bug 1113056](https://bugzilla.mozilla.org/show_bug.cgi?id=1113056) to get this.
Currently, this code fails for me on the delete because it sends a 404 with the error: "No matching user setting found."